src/views/token-swap/components/amount-box/amount-box.view.jsx

```diff
@@ -4,89 +4,63 @@ import PropTypes from 'prop-types'
-import AmountInput from '../../../shared/amount-input/amount-input'
+import AmountInput from '../../../shared/amount-input/amount-input.view'
 import Dropdown from '../dropdown/dropdown.view'
 import { getTokenIcon } from '../../../../utils/tokens'
 import {
   getTokenAmountInPreferredCurrency,
   getFixedTokenAmount
 } from '../../../../utils/currencies'
 
+export const AmountBoxPosition = {
+  TO: 'to',
+  FROM: 'from'
+}
+
 function AmountBox ({
   account,
   preferredCurrency,
   fiatExchangeRates,
   value,
-  amount,
-  onInputChange,
   position,
   accounts,
-  setToken,
-  setDropdown,
-  isDropdownActive
+  isDropdownActive,
+  onInputChange,
+  onTokenChange,
+  onActiveDropdownChange
 }) {
   const classes = useAmountBoxStyles()
   const balance = getFixedTokenAmount(account?.balance, account?.token.decimals)
-  const setMax = () => {
-    onInputChange({
-      target: { value: balance }
-    })
-  }
-  const switchDropdown = () => {
-    const isActive = isDropdownActive ? '' : position
-    setDropdown(isActive)
-  }
   const convertValue = getTokenAmountInPreferredCurrency(
     value,
     account?.token.USD,
     preferredCurrency,
     fiatExchangeRates
   )
-  const Icon = getTokenIcon(account?.token.symbol)
-  const renderName = () => {
-    if (account) {
-      return (
-        <div className={classes.tokenName}>
-          <Icon className={classes.tokenIcon} /> {account.token.symbol}
-        </div>
-      )
-    }
-    return 'Select token'
-  }
-  const renderBalance = () => {
-    if (account) {
-      return (
-        <p className={classes.convertedText}>
-          Balance: {balance} {account.token.symbol}{' '}
-          {position === 'from' && (
-            <span className={classes.maxBtn} onClick={setMax}>
-              Max
-            </span>
-          )}
-        </p>
-      )
-    }
-    return <div />
+
+  const setMax = () => {
+    onInputChange({ target: { value: balance } })
   }
-  const renderDropdown = () => {
-    if (isDropdownActive) {
-      return (
-        <Dropdown
-          close={switchDropdown}
-          accounts={accounts}
-          setToken={setToken}
-          position={position}
-        />
-      )
-    }
-    return null
+
+  const switchDropdown = () => {
+    const isActive = isDropdownActive ? '' : position
+    onActiveDropdownChange(isActive)
   }
+
+  const Icon = getTokenIcon(account?.token.symbol)
+
   return (
     <div className={classes.frame}>
       <div className={classes.box}>
         <div className={classes.row}>
           <div className={classes.selectorBox} onClick={switchDropdown}>
-            {renderName()} <AngleDown className={classes.angleColor} />
+            {account
+              ? (
+                <div className={classes.tokenName}>
+                  <Icon className={classes.tokenIcon} /> {account.token.symbol}
+                </div>
+                )
+              : 'Select token'} <AngleDown className={classes.angleColor} />
```

Comment:
Can we wrap the `'Select token'` string into a `p` element? It's going to be more correct.

Reply:
Maybe a span if you want to wrapper it but it's in the same line that angledown and not need for more style

Reply:
Well, it's not for styling but for write more semantically correct HTML. A span is enough.